### PR TITLE
Feature s rgb consistency throughout pipelines

### PR DESCRIPTION
**Pull request by LaisoEmilio**
sRGB was not consistent throughout the pipelines (deferred, hybrid and full rt). This branch will fix that, as well as fixing a problem with sRGB format handling in the texture pool

### CODE REVIEW

File: src/render_tasks/d3d12_deferred_composition.cpp [hunk 1]
```diff
@@ -200,7 +206,7 @@ namespace wr
 				// Output UAV
 				{
 					auto rtv_out_uav_handle = data.out_srv_uav_allocation.GetDescriptorHandle(COMPILATION_EVAL(rs_layout::GetHeapLoc(params::deferred_composition, params::DeferredCompositionE::OUTPUT)));
-					std::vector<Format> formats = { Format::R8G8B8A8_UNORM };
+					std::vector<Format> formats = { Format::R32G32B32A32_FLOAT };
```
Comment (DBouma) [inline src/render_tasks/d3d12_deferred_composition.cpp]: Is 32 bit required? I believe 16bit is enough right?

File: src/render_tasks/d3d12_deferred_composition.cpp [hunk 2]
```diff
@@ -277,7 +283,7 @@ namespace wr
 			RenderTargetProperties::FinishedResourceState(ResourceState::COPY_SOURCE),
 			RenderTargetProperties::CreateDSVBuffer(false),
 			RenderTargetProperties::DSVFormat(Format::UNKNOWN),
-			RenderTargetProperties::RTVFormats({ Format::R8G8B8A8_UNORM }),
+			RenderTargetProperties::RTVFormats({ Format::R32G32B32A32_FLOAT }),
```
Comment (DBouma) [inline src/render_tasks/d3d12_deferred_composition.cpp]: Consider format as previously commented

File: src/render_tasks/d3d12_raytracing_task.hpp [hunk 3]
```diff
@@ -251,6 +252,13 @@ namespace wr
 					d3d12::SetRTShaderSRV(cmd_list, 0, COMPILATION_EVAL(rs_layout::GetHeapLoc(params::full_raytracing, params::FullRaytracingE::SKYBOX)), skybox_t);
 				}
 
+        // Get brdf lookup texture
+        if (scene_graph.m_skybox.has_value())
+        {
+          auto brdf_lut_text = static_cast<d3d12::TextureResource*>(n_render_system.m_brdf_lut.value().m_pool->GetTexture(n_render_system.m_brdf_lut.value().m_id));
```
Comment (DBouma) [inline src/render_tasks/d3d12_raytracing_task.hpp]: "brdf_lut_text" this name does not make sense, what I'm reading right now is "brdf lookup texture text" , pretty sure you are not referring to text.

File: src/render_tasks/d3d12_rt_hybrid_task.hpp [hunk 4]
```diff
@@ -397,7 +398,7 @@ namespace wr
 			RenderTargetProperties::FinishedResourceState(ResourceState::COPY_SOURCE),
 			RenderTargetProperties::CreateDSVBuffer(false),
 			RenderTargetProperties::DSVFormat(Format::UNKNOWN),
-			RenderTargetProperties::RTVFormats({ Format::R8G8B8A8_UNORM }),
+			RenderTargetProperties::RTVFormats({ Format::R32G32B32A32_FLOAT }),
```
Comment (DBouma) [inline src/render_tasks/d3d12_rt_hybrid_task.hpp]: Consider format as previously commented

File: resources/shaders/raytracing.hlsl [hunk 5]
```diff
@@ -266,8 +268,9 @@ void ClosestHitEntry(inout HitInfo payload, in MyAttributes attr)
 		s0,
 		uv);
 
-	float3 albedo = output_data.albedo;
-	float roughness = output_data.roughness;
+	float3 albedo = pow(output_data.albedo, 2.2f);
+  //float3 albedo = output_data.albedo;
+  float roughness = output_data.roughness;
```
Comment (DBouma) [inline resources/shaders/raytracing.hlsl]: There seem to be a lot of inconsistant tabbing problems in all of your changes, double check if they are formatted correctly or if this is just the web viewer giving these weird spacings.

File: src/d3d12/d3d12_resource_pool_texture.cpp [hunk 6]
```diff
@@ -589,21 +593,23 @@ namespace wr
...
-		{
-			adjusted_format = static_cast<wr::Format>(metadata.format);
+    adjusted_format = static_cast<wr::Format>(metadata.format);
 
-			if (d3d12::CheckSRGBFormat(adjusted_format))
-			{
-				adjusted_format = d3d12::RemoveSRGB(adjusted_format);
-			}
-		}
+		//if (srgb)
```
Comment (VZout) [inline src/d3d12/d3d12_resource_pool_texture.cpp]: REEEEEEEEEEEEEEEEEEEEEEEEEEE

File: src/d3d12/d3d12_resource_pool_texture.cpp [hunk 7]
```diff
@@ -511,21 +513,23 @@ namespace wr
...
```
Comment (VZout) [inline src/d3d12/d3d12_resource_pool_texture.cpp]: REEEEEEEEEEEEEEE

File: src/d3d12/d3d12_resource_pool_texture.cpp [hunk 8]
```diff
@@ -432,19 +432,21 @@ namespace wr
...
-			adjusted_format = static_cast<wr::Format>(metadata.format);
-
-			if (d3d12::CheckSRGBFormat(adjusted_format))
-			{
-				adjusted_format = d3d12::RemoveSRGB(adjusted_format);
-			}
-		}
+    adjusted_format = static_cast<wr::Format>(metadata.format);
+
+		//if (srgb)
```
Comment (VZout) [inline src/d3d12/d3d12_resource_pool_texture.cpp]: source control 🌈

**Pull request merged**